File: slidingWindow/longestSubStringWithoutRepeatingCharacters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        numberOfAscii = 128
        indArr = [0] * numberOfAscii
        l, r = 0, 0
        maxSubStringLen = 0
        strLen = len(s)
        while r < strLen:
            if indArr[ord(s[r])]:
                indArr[ord(s[l])] -= 1
                l += 1
            else:
                indArr[ord(s[r])] += 1
                r += 1
                maxSubStringLen = max(maxSubStringLen, r - l)
        return maxSubStringLen

File: slidingWindow/longestSubStringWithoutRepeatingCharacters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int res = 0;
        int l = 0;
        int r = 0;
        vector<int> freq(128, 0);
        while (r < s.size()){
            while(freq[s[r]]){
                freq[s[l]]--;
                l++;
            }
            freq[s[r]]++;
            r++;
            res = max(res, r - l);
        }
        return res;
    }
};
